docs/Blog-Post.md

```diff
@@ -0,0 +1,276 @@
+and scale, as well as strong guarantees about reproducibility.
+
+Levanter is a [Jax](https://github.com/google/jax)-based codebase for training foundation models that is designed to be
+flexible, modular, and accessible, while still being performant and scalable. In particular, Levanter is designed to be
+able to train models on a variety of different hardware, including GPUs, TPUs, and TPU pods. Levanter also offers strong
+guarantees about reproducibility, and we have released checkpoints for a number of models trained with Levanter.
+Haliax is a named tensor library that we developed to make it easier to write legible, composable code while
+still maintaining efficiency and scalability.
+
+Today, we're releasing the first version of Levanter and Haliax, along with several models trained with Levanter. We
+hope that these libraries will be useful to the community, and we look forward to seeing what people do with them.
+
+## The Landscape of Foundation Model Training Frameworks
+
+Numerous foundation model training frameworks exist in the community, each with its strengths and focus.
+For large language models (LLMs) (the focus of this release),
+the most well-known in the open source community is probably NVIDIA's PyTorch-based [Megatron-LM](https://github.com/NVIDIA/Megatron-LM),
+and its many derivatives, including EleutherAI's [GPT-NeoX](https://github.com/EleutherAI/gpt-neox) codebase. MosaicML
+has recently released [LLM Foundry](https://github.com/mosaicml/llm-foundry), which we used to train
+[BioMedLM](https://crfm.stanford.edu/2022/12/15/biomedlm.html) last autumn.
+
+In the Jax community, there are a number of libraries popping up. Google has released [T5X](https://github.com/google-research/t5x)
+and [MaxText](https://github.com/google/maxtext). There are also a number of independent libraries, including [EasyLM](https://github.com/young-geng/EasyLM)
+and [JaxSeq](https://github.com/Sea-Snell/JAXSeq), both of which are based on [Flax](https://github.com/google/flax/)
+and modified libraries from [Hugging Face Transformers](https://github.com/huggingface/transformers/). Salesforce has
+released the Haiku-based [Jaxformer](https://github.com/salesforce/jaxformer).
+
+Despite the wide array of existing frameworks, when we started, we found that none of them fully addressed our needs.
+At CRFM, we wanted to achieve four goals:
+
+* **Legibility**: We wanted to be able to write code that was easy to read and understand, and that could be composed easily.
+* **Reproducibility**: We wanted to be able to reproduce our results exactly, even in the presence of preemption and
+  restarting from checkpoints.
+* **Scalability**: We wanted to be able to fully utilize the Cloud TPU resources we were given as part of Google's TPU Research Cloud program, as well as our own GPU resources.
+* **Efficiency**: We wanted to achieve the other three goals without sacrificing (much) efficiency.
```

Comment:
Can these be combined into one?  Rule of 3s...

Reply:
i'll try...